src/tasks/models/tasksManager.ts

```diff
@@ -0,0 +1,127 @@
+import { Logger } from '@map-colonies/js-logger';
+import { inject, injectable } from 'tsyringe';
+import config from 'config';
+import { IUpdateJobBody, OperationStatus } from '@map-colonies/mc-priority-queue';
+import { NotFoundError } from '@map-colonies/error-types';
+import { getGpkgFullPath, getGpkgRelativePath } from '../../common/utils';
+import { SERVICES } from '../../common/constants';
+import { JobManagerWrapper } from '../../clients/jobManagerWrapper';
+import { ICallbackData, ICallbackDataBase, IJobParameters, IJobStatusResponse, JobResponse } from '../../common/interfaces';
+import { CallbackClient } from '../../clients/callbackClient';
+import { getFileSize } from '../../common/utils';
+import { CreatePackageManager } from '../../createPackage/models/createPackageManager';
+
+export interface ITaskStatusResponse {
+  percentage: number | undefined;
+  status: OperationStatus;
+}
+
+@injectable()
+export class TasksManager {
+  private readonly gpkgsLocation: string;
+  private readonly downloadServerUrl: string;
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(JobManagerWrapper) private readonly jobManagerClient: JobManagerWrapper,
+    @inject(CallbackClient) private readonly callbackClient: CallbackClient,
+    @inject(CreatePackageManager) private readonly packageManager: CreatePackageManager
+  ) {
+    this.gpkgsLocation = config.get<string>('gpkgsLocation');
+    this.downloadServerUrl = config.get<string>('downloadServerUrl');
+  }
+
+  public async getJobsByTaskStatus(): Promise<IJobStatusResponse> {
+    const jobs = await this.jobManagerClient.getJobsStatus();
+    const completedJobs = jobs?.filter((job) => job.completedTasks === job.taskCount);
+    const failedJobs = jobs?.filter((job) => job.failedTasks === job.taskCount);
+    const jobsStatus = {
+      completedJobs: completedJobs,
+      failedJobs: failedJobs,
+    };
+    return jobsStatus;
+  }
+
+  public async getTaskStatusByJobId(jobId: string): Promise<ITaskStatusResponse> {
+    const tasks = await this.jobManagerClient.getTasksByJobId(jobId);
+
+    if (tasks.length === 0) {
+      throw new NotFoundError(`No tasks were found for jobId: ${jobId}`);
+    }
+    const task = tasks[0];
+    const statusResponse: ITaskStatusResponse = {
+      percentage: task.percentage,
+      status: task.status,
+    };
+    return statusResponse;
+  }
+
+  public async sendCallbacks(job: JobResponse, expirationDate: Date, errorReason?: string): Promise<ICallbackDataBase | undefined> {
+    try {
+      this.logger.info(`Sending callback for job: ${job.id}`);
+      const packageName = job.parameters.fileName;
+      const fileRelativePath = getGpkgRelativePath(packageName);
+      const fileUri = `${this.downloadServerUrl}/downloads/${fileRelativePath}`;
```

Comment:
why do you calculate "packageFullPath" or put URI if success == false?

Reply:
Fix tests

Reply:
URI fixed